polaris/benchmark/predictions/_base.py

```diff
@@ -0,0 +1,129 @@
+
+        if isinstance(vals, (list, np.ndarray)):
+            return {"test": cls._set_col_name(vals, target_cols)}
+        elif cls._is_multi_task_single_test_set(vals, target_cols):
+            return {"test": vals}
+        elif cls._is_single_task_multi_test_set(vals, target_cols):
+            return {test_set_name: cls._set_col_name(v, target_cols) for test_set_name, v in vals.items()}
+        else:
+            cls._give_feedback(vals)
+        return vals
+
+    @classmethod
+    def _convert_lists_to_numpy_arrays(
+        cls, predictions: dict[str, dict[str, ListOrArrayType]]
+    ) -> PredictionsType:
+        """Converts all values in the predictions dictionary to numpy arrays."""
+
+        def convert_to_numpy(v):
+            if isinstance(v, (list, np.ndarray)):
+                return np.array(v)
+            elif isinstance(v, dict):
+                return {k: convert_to_numpy(v) for k, v in v.items()}
+
+        return convert_to_numpy(predictions)
+
+    @classmethod
+    def _is_multi_task_single_test_set(cls, vals, target_cols) -> bool:
+        """Check if the given values are for a multiple-task benchmark with a single
+        test set. This is inferred by comparing the target names with the keys of the
+        given data. If all keys in the given data match the target column names, we
+        assume they are target names (as opposed to test set names for a single-task,
+        multiple test set benchmark)."""
+        return isinstance(vals, dict) and all(k in target_cols for k in vals.keys())
+
+    @classmethod
+    def _is_single_task_multi_test_set(cls, vals, target_cols) -> bool:
+        """Check if the given values are for a single-task benchmark with multiple
+        test sets. This is inferred by comparing the target names with the keys of the
+        given data. If there is a single target column but more than one dictionary entry,
+        and none of the dictionary keys match the target column name, we assume the
+        dictionary keys are test set names and the values are the predictions for each test set.
+        """
+        return (
+            len(target_cols) == 1
+            and isinstance(vals, dict)
+            and len(vals.keys()) > 1
+            and all(k != target_cols[0] for k in vals.keys())
+        )
+
+    @classmethod
+    def _set_col_name(cls, vals, target_cols: list[str]):
+        if isinstance(vals, dict):
+            return vals
+        else:
+            return {target_cols[0]: vals}
+
+    @classmethod
+    def _give_feedback(cls, vals):
+        for test_set, targets in vals.items():
+            if not isinstance(targets, dict):
+                raise ValueError(
+                    f"Invalid structure for test set '{test_set}'. "
+                    "Expected a dictionary of {col_name: predictions}"
+                )
+            for target, predictions in targets.items():
+                if not (
+                    isinstance(predictions, np.ndarray)
```

Comment:
**Suggestion:** Should we also check if the NumPy array is numeric?

Reply:
Actually, a related comment: I could imagine cases in which predictions can be non-numeric, e.g. if you predict the outcome of a chemical reaction you would likely pass the prediction as a SMILES string. We don't host any such benchmarks right now, but if we want to make sure current versions of the client can support such future benchmarks, maybe we shouldn't add this check?

Reply:
The `ListOrArrayType` should only allow ints or floats, the idea was to only support numeric predictions. But yeah this would be the place to update if/when we want to support non-numeric predictions. I can separate out a named method to make it more clear that this is meant to encapsulate validating specifically the prediction values.